packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -145,17 +146,25 @@ class IOSDeploy {
       '--id',
       deviceId,
       '--exists',
+      '--timeout', // If the device is not connected, ios-deploy will wait forever.
+      '10',
       '--bundle_id',
       bundleId,
     ];
     final RunResult result = await _processUtils.run(
       launchCommand,
       environment: iosDeployEnv,
     );
+    // Device successfully connected, but app not installed.
```

Comment:
Does ios-deploy handle cases where the bundle identifiers are the same but the contents are different?

Reply:
No, I think `--exists` just checks the bundle ID.

Reply:
Ahh, I see that for iOS `islatestBuildInstalled is always false. https://github.com/flutter/flutter/blob/1ec2a43672c1d7a5dd9ef3b295b1931956e41782/packages/flutter_tools/lib/src/ios/devices.dart#L145

NVM, I was slightly confused between the two